src/number_theory/bernoulli_polynomials.lean

```diff
@@ -241,3 +244,128 @@ begin
 end
 
 end polynomial
+
+/-
+## Fourier coefficients of the Bernoulli polynomials
+-/
+open_locale real interval
+open complex measure_theory set interval_integral
+
+section bernoulli_fun_props
+/-! Simple properties of the Bernoulli polynomial, as a function `ℝ → ℝ`. -/
+
+/-- The function `x ↦ Bₖ(x) : ℝ → ℝ`. -/
+def bernoulli_fun (k : ℕ) (x : ℝ) : ℝ :=
+(polynomial.map (algebra_map ℚ ℝ) (polynomial.bernoulli k)).eval x
+
+local notation `B` := bernoulli_fun
+
+lemma bernoulli_fun_eval_zero (k : ℕ): B k 0 = bernoulli k :=
+by rw [bernoulli_fun, polynomial.eval_zero_map, polynomial.bernoulli_eval_zero, eq_rat_cast]
+
+lemma bernoulli_fun_endpoints_eq_of_ne_one {k : ℕ} (hk : k ≠ 1) : B k 0 = B k 1 :=
+by rw [bernoulli_fun_eval_zero, bernoulli_fun, polynomial.eval_one_map,
+  polynomial.bernoulli_eval_one, bernoulli_eq_bernoulli'_of_ne_one hk, eq_rat_cast]
+
+lemma bernoulli_fun_eval_one (k : ℕ) : B k 1 = B k 0 + ite (k = 1) 1 0 :=
+begin
+  rw [bernoulli_fun, bernoulli_fun_eval_zero, polynomial.eval_one_map,
+    polynomial.bernoulli_eval_one],
+  split_ifs,
+  { rw [h, bernoulli_one, bernoulli'_one, eq_rat_cast, rat.cast_div, rat.cast_div, rat.cast_neg,
+    rat.cast_bit0, rat.cast_one], -- somehow norm_cast doesn't handle this?
+    ring },
+  { rw [bernoulli_eq_bernoulli'_of_ne_one h, add_zero, eq_rat_cast], }
+end
+
+lemma has_deriv_at_bernoulli_fun (k : ℕ) (x : ℝ) : has_deriv_at (B k)
+  (k * bernoulli_fun (k - 1) x) x :=
+begin
+  convert ((polynomial.bernoulli k).map $ algebra_map ℚ ℝ).has_deriv_at x using 1,
+  simp only [bernoulli_fun, polynomial.derivative_map, polynomial.derivative_bernoulli k,
+    polynomial.map_mul, polynomial.map_nat_cast, polynomial.eval_mul, polynomial.eval_nat_cast],
+end
+
+lemma antideriv_bernoulli_fun (k : ℕ) (x : ℝ) :
+  has_deriv_at (λ x, (B (k + 1) x) / (k + 1)) (B k x) x :=
+begin
+  convert (has_deriv_at_bernoulli_fun (k + 1) x).div_const _,
+  field_simp [nat.cast_add_one_ne_zero k],
+  ring,
+end
+
+lemma integral_bernoulli_fun_eq_zero (k : ℕ) (hk : 1 ≤ k) :
+  ∫ (x : ℝ) in 0..1, B k x = 0 :=
+begin
+  rw integral_eq_sub_of_has_deriv_at (λ x hx, antideriv_bernoulli_fun k x)
+    ((polynomial.continuous _).interval_integrable _ _),
+  dsimp only,
+  rw bernoulli_fun_eval_one,
+  split_ifs,
+  { exfalso, linarith, }, { simp },
+end
+
+end bernoulli_fun_props
+
+/-! Compute the Fourier coefficients of the Bernoulli functions via integration by parts. -/
+section bernoulli_fourier_coeffs
+
+local notation `e` := (λ n:ℤ, (fourier n : C(unit_add_circle, ℂ)))
+
+/-- The `n`-th Fourier coefficient of the `k`-th Bernoulli function. -/
+def bernoulli_fourier_coeff (k : ℕ) (n : ℤ) : ℂ :=
```

Comment:
You are defining this, but all the lemmas below are expressed in terms of bare integrals. Can you switch the statements of the lemmas to use the definition (or remove the definition)?